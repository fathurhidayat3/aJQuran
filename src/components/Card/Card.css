.Card {
  padding: 1rem;

  background: white;
  border-radius: 4px;
  box-shadow: 0 2px 7px 0 #dcdde1;

  display: flex;
  flex-direction: column;

  text-decoration: none;

  /* cursor: pointer; */
}
.Card:not(:last-child) {
  margin-bottom: 13px;
}

.Card-title {
  margin-bottom: 10px;

  font-family: Ubuntu;
  font-weight: bold;
  font-size: 1rem;
  color: #2f3640;
}
.Card-title-helper {
  font-weight: normal;
  color: #dcdde1;
}
.Card-title-right {
  font-size: 1.2rem;
  font-family: 'Harmattan';
  font-weight: normal;
  color: #44bd32;
  line-height: 1.2rem;

  float: right;
}

.Card-footer {
  font-family: Ubuntu;
  font-size: 0.8rem;
  color: #999;
  line-height: 1rem;
}

.Card.active {
  background: #273c75;
  /* background: linear-gradient(to bottom, #0097e6 75%, #192a56 100%);
  background: -moz-linear-gradient(to bottom, #0097e6 75%, #192a56 100%);
  background: -webkit-linear-gradient(to bottom, #0097e6 75%, #192a56 100%); */
}

.Card.active .Card-title {
  color: white;
}
.Card.active .Card-title-helper {
  color: #f1f1f1;
}
.Card.active .Card-title-right {
  color: #fbc531;
}

.Card.active .Card-footer {
  color: #e1b12c;
}