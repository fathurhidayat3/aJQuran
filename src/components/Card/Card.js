import React, { Component } from "react";
import { NavLink } from "react-router-dom";

import "./Card.css";

class Card extends Component {
  constructor(props) {
    super(props);
  }

  onCardClick(number) {
    this.props.onCardClick(number);
  }

  render() {
    let { surat } = this.props;

    return (
      <NavLink
        className="Card"
        to={`/quran/${surat.number}`}
        activeStyle={{ className: "active" }}
        onClick={() => this.onCardClick(surat.number)}
      >
        <div className="Card-title">
          {surat.englishName}
          <span className="Card-title-helper"> ({surat.revelationType})</span>
          <span className="Card-title-right">{surat.name}</span>
        </div>

        <div className="Card-footer">{surat.numberOfAyahs} ayat</div>
      </NavLink>
    );
  }
}

export default Card;
