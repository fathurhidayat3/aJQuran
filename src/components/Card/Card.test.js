import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

it('without component', () => {  
  const CardComponent = shallow(<Card />);
  expect(CardComponent).toMatchSnapshot();
});