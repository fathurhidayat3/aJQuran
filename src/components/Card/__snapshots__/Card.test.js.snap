// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`without component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeClassName": "active",
      "children": Array [
        <div
          className="Card-title"
        >
          <span
            className="Card-title-right"
          />
        </div>,
        <div
          className="Card-footer"
        />,
      ],
      "className": "Card active",
      "onClick": [Function],
      "smooth": true,
      "to": "/quran/undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            null,
            <span
              className="Card-title-right"
            />,
          ],
          "className": "Card-title",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "Card-title-right",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "Card-footer",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "children": Array [
          <div
            className="Card-title"
          >
            <span
              className="Card-title-right"
            />
          </div>,
          <div
            className="Card-footer"
          />,
        ],
        "className": "Card active",
        "onClick": [Function],
        "smooth": true,
        "to": "/quran/undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              null,
              <span
                className="Card-title-right"
              />,
            ],
            "className": "Card-title",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "Card-title-right",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "Card-footer",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
